foam_controlwrapper/foam_mesh.py

```diff
@@ -0,0 +1,772 @@
+    _uuidToFoamLabel : dictionary
+        Mapping from uuid to OpenFoam label number
+    _foamCellLabelToUuid : dictionary
+        Mapping from OpenFoam cell label number to uuid
+    _foamFaceLabelToUuid : dictionary
+        Mapping from  OpenFoam face label number to uuid
+    _foamEdgeLabelToUuid : dictionary
+        Mapping from  OpenFoam edge label number to uuid
+    _foamPointLabelToUuid : dictionary
+        Mapping from  OpenFoam point label number to uuid
+
+    """
+
+    def __init__(self, name, mesh=None, path=None):
+        self.name = name
+        self.data = dc.DataContainer()
+        self.time = str(0)
+        self._uuidToFoamLabel = {}
+        self._foamCellLabelToUuid = {}
+        self._foamFaceLabelToUuid = {}
+        self._foamEdgeLabelToUuid = {}
+        self._foamPointLabelToUuid = {}
+
+        if path:
+            self.path = path
+        else:
+            self.path = os.path.join(tempfile.mkdtemp(), name)
+        if mesh:
+            # generate uuid mapping
+            label = 0
+            for point in mesh.iter_points():
+                uid = point.uid
+                self._uuidToFoamLabel[uid] = label
+                self._foamPointLabelToUuid[label] = uid
+                label += 1
+
+            label = 0
+            for edge in mesh.iter_edges():
+                uid = edge.uid
+                self._uuidToFoamLabel[uid] = label
+                self._foamEdgeLabelToUuid[label] = uid
+                label += 1
+
+            label = 0
+            for face in mesh.iter_faces():
+                uid = face.uid
+                self._uuidToFoamLabel[uid] = label
+                self._foamFaceLabelToUuid[label] = uid
+                label += 1
+
+            label = 0
+            for cell in mesh.iter_cells():
+                uid = cell.uid
+                self._uuidToFoamLabel[uid] = label
+                self._foamCellLabelToUuid[label] = uid
+                label += 1
+
+            # find out boundary patches
+            patchNameFacesMap = {}
+            facePoints = []
+            for face in mesh.iter_faces():
+                if CUBA.LABEL in face.data:
+                    boundary = 'boundary' + str(face.data[CUBA.LABEL])
+                    if boundary not in patchNameFacesMap:
+                        patchNameFacesMap[boundary] = []
+                    patchNameFacesMap[boundary].append(
+                        self._uuidToFoamLabel[face.uid])
+
+                # make compressed list of faces points
+                facePoints.append(len(face.points))
+                for puid in face.points:
+                    if type(puid) is not numpy.string_:
+                        facePoints.append(self._uuidToFoamLabel[puid])
+                    else:
+                        facePoints.append(
+                            self._uuidToFoamLabel[uuid.UUID(puid,
+                                                            version=4)])
```

Comment:
Please add a comment in the code pointing to the related github issue when workarounds have to take place because of bugs. However, it is not clear why ``FileMesh`` effects the implementation of the openfoam adapter.